node/sync.go

```diff
@@ -229,6 +231,19 @@ func (d *Pegnetd) SyncBlock(ctx context.Context, tx *sql.Tx, height uint32) erro
 			return err
 		}
 
+		// Before we apply any balance changes, we will snapshot the balances at the START of the block.
+		// This means the balances are the same as the end of the block n-1.
+		// This activation is nested in the activation that has the rates
+		// TODO: if the height has no rates, what do we do?
+		//		We need to handle the no rates case. Miners could avoid mining this last block.
+		// 		Can we use the last valid rates?
+		if height >= V20HeightActivation && height%pegnet.SnapshotRate == 0 { // Put in act height
+			err := d.SnapshotPayouts(tx, fLog, rates, height, dblock.Timestamp)
```

Comment:
@sigrlami , is it still todo option? i think you should add validation codes if `rates` is nil or not.
please compare `rates` is nil here, and if it is `nil`, use last valid rates from sqlite.

Reply:
added reading from previous height, skips payout for more